anu_lms.module

```diff
@@ -32,32 +32,31 @@ function anu_lms_form_node_module_delete_form_alter(&$form) {
  */
 function anu_lms_node_delete(NodeInterface $node) {
   // Collects leftover contents IDs.
-  switch ($node->getType()) {
-    case 'course':
-      $nids = \Drupal::entityQuery('node')
-        ->condition('type', 'module')
-        ->condition('field_module_course.target_id', $node->id())
-        ->execute();
-      break;
+  if ($node->getType() == 'course') {
 
-    case 'module':
-      $lessons_nids = \Drupal::entityQuery('node')
-        ->condition('type', 'module_lesson')
-        ->condition('field_module_lesson_module.target_id', $node->id())
-        ->execute();
+    // Collect module paragraphs IDs.
+    $modules = $node->get('field_course_module')->referencedEntities();
```

Comment:
just a note for future:
`referencedEntities();` method is might affect performance, so in some cases it's much better to use `\Drupal::entityQuery` instead.

Reply:
I am aware of `entityQuery`, thanks. For this particular case, we need entities instead of ids, so even if entity Query is easier I would have to loads entities later. So for this case, I think the performance will be more or less similar.